src/tools/archiveTools.cpp

```diff
@@ -105,5 +105,13 @@ zim::Entry getFinalEntry(const zim::Archive* const archive, const zim::Entry& en
   return archive->getEntryByPath(entry.getItem(true).getPath());
 }
 
+zim::Entry getEntryFromPath(const zim::Archive* const archive, const std::string& path)
+{
+  if (path.empty() || path == "/") {
+    return archive->getMainEntry();
+  }
+  return archive->getEntryByPath(path);
+}
+
```

Comment:
I don't think we should add it to libzim because this is just a special user(libkiwix) need to get the `mainentry` if the requested entry is not found. Like you mentioned, there can be cases which as a library should not bother libzim. 

The new behaviour you suggested sounds good, I will make the change.